contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -56,12 +56,14 @@ contract BosonVoucherBase is IBosonVoucher, BeaconClientBase, OwnableUpgradeable
     // Tell if preminted voucher has already been _committed
     mapping(uint256 => bool) private _committed;
 
+    mapping(uint256 => address) private _preMintOwners;
```

Comment:
Storing address per every `tokenId` does not makes much sense, since it invalidates all minting gas savings that were introduced with ranges. The main reasons to introduce ranges was to avoid storing the same address all over the place.
If we wanted to store address per token id it would be better to just scrap everything we have, and make a simple loop where we call existing OZ `_mint` method, which would be more efficient and less error prone (no need to store ranges, no need to do binary search).
I run limit estimation and now max preminted vouchers (in single tx) dropped to arround 1.000, while before it was > 10.000.

It would be better to just add `owner" field to existing `Range` struct.

Reply:
but then it would be necessary to create a `reserveRangeTo` method